test/csuite/timestamp_abort/main.c

```diff
@@ -1144,11 +1230,20 @@ recover_and_verify(uint32_t backup_index)
     if (backup_index == 0) {
         testutil_wiredtiger_open(opts, WT_HOME_DIR, NULL, &my_event, &conn, true, false);
         printf("Connection open and recovery complete. Verify content\n");
+        /* Compare against the copy of the home directory just before recovery. */
+        if (use_backups) {
+            printf("==== Verify saved dir against backup source ====\n");
+            testutil_check(__wt_snprintf(buf, sizeof(buf), "%s.SAVE/%s", home, WT_HOME_DIR));
+            testutil_verify_src_backup(conn, buf, WT_HOME_DIR, NULL);
+            printf("==== DONE Verify saved dir against backup source ====\n");
+        }
     } else {
         testutil_check(__wt_snprintf(buf, sizeof(buf), BACKUP_BASE "%" PRIu32, backup_index));
         testutil_system("rm -rf check; cp -rf %s check", buf);
         testutil_wiredtiger_open(opts, "check", NULL, &my_event, &conn, true, false);
     }
+    if (use_backups)
```

Comment:
This should be moved up to the `if (backup_index == 0) {` branch, because that's when we are doing `recover_and_verify` on the main database, not on a backup. (This could become a problem if we ever decide to to support both options for database verification as I mentioned in my earlier comment.)